quinn/src/streams.rs

```diff
@@ -573,26 +582,23 @@ where
         cx: &mut Context,
         buf: &mut [u8],
     ) -> Poll<io::Result<usize>> {
-        Poll::Ready(Ok(
-            ready!(RecvStream::poll_read(self.get_mut(), cx, buf))?.unwrap_or(0)
-        ))
+        let mut buf = ReadBuf::new(buf);
+        ready!(RecvStream::poll_read(self.get_mut(), cx, &mut buf))?;
+        Poll::Ready(Ok(buf.filled().len()))
     }
 }
 
 impl<S> tokio::io::AsyncRead for RecvStream<S>
 where
     S: proto::crypto::Session,
 {
-    unsafe fn prepare_uninitialized_buffer(&self, _: &mut [MaybeUninit<u8>]) -> bool {
-        false
-    }
-
     fn poll_read(
         self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-        buf: &mut [u8],
-    ) -> Poll<io::Result<usize>> {
-        AsyncRead::poll_read(self, cx, buf)
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
+        let _ = ready!(RecvStream::poll_read(self.get_mut(), cx, buf))?;
```

Comment:
Is the `let _` here necessary, given that `RecvStream::poll_read` has a `()` success type?

Reply:
Doesn't appear to be!